src/app/shared/filesystem/DirectoryTree.ts

```diff
@@ -238,7 +238,7 @@ export class DirectoryTree {
    * @memberof DirectoryTree
    */
   private removeEmptyFolders(node: DirectoryNode): void {
-    if (!node.getChildren().length && !node.getFiles().length) {
+    if (!node && !node.getChildren().length && !node.getFiles().length) {
```

Comment:
@nrwinner The first check here checks if `node` is falsy. If `node` is falsy, wouldn't that cause the subsequent checks and code within the if block to fail. Am I missing something?